spec/policy_machine_storage_adapters/active_record_spec.rb

```diff
@@ -107,6 +107,207 @@
           expect(user.linked?(pm2_user)).to eq true
         end
       end
+
+      describe '#bulk_persist' do
+        let(:pm) { PolicyMachine.new(name: 'AR PM', storage_adapter: PolicyMachineStorageAdapter::ActiveRecord) }
+        let(:user) { pm.create_user('alice') }
+        let(:caffeinated) { pm.create_user_attribute('caffeinated') }
+        let(:decaffeinated) { pm.create_user_attribute('decaffeinated') }
+
+        describe 'policy element behavior' do
+          it 'deletes a policy element that has been created and then deleted in a persistence buffer' do
+            user, attr = pm.bulk_persist do
+              user = pm.create_user('alice')
+              attr = pm.create_user_attribute('caffeinated')
+              user.delete
+
+              [user, attr]
+            end
+
+            expect(pm.user_attributes).to eq [attr]
+            expect(pm.users).to be_empty
```

Comment:
Could also do/check `user.persisted?`. Not saying it's better or worse than this way.